### PR TITLE
Top level Unreal Engine and plugin

### PR DESCRIPTION
**Pull request by tustanivsky**
Adds documentation with examples and screenshots for the upcoming Sentry Unreal Engine plugin.

### CODE REVIEW

File: src/docs/contributing/platforms/index.mdx [hunk 1]
```diff
@@ -110,8 +110,6 @@ If you wanted to make hide the content for a platform by default, but enable it
 
 ```markdown
 ---
-supported:
-  - native.ue4
 notSupported:
   - native
```
Comment (imatwawana) [inline src/docs/contributing/platforms/index.mdx:114]:
```suggestion
supported: 
  - native.wasm
notSupported:
  - native
```

File: src/platforms/unreal/setup-crashreport.mdx [hunk 2]
```diff
@@ -81,6 +78,52 @@ is enough.
 
 </Alert>
 
+## Configure the Crash Reporter Attributes {#configure-attributes}
+
+Basic attributes can be reconfigured by providing a special game data to the
+crash reporter by setting the `__sentry` game data key to a JSON value containing
+Sentry specific attributes:
```
Comment (imatwawana) [inline src/platforms/unreal/setup-crashreport.mdx:85]:
```suggestion
Basic attributes can be reconfigured by providing special game data to the
crash reporter. Do this by setting the `__sentry` game data key to a JSON value containing
Sentry specific attributes:
```

File: src/platforms/unreal/setup-crashreport.mdx [hunk 3]
```diff
@@ -81,6 +78,52 @@ is enough.
...
+    FString jsonConfig;
+    TSharedRef<TJsonWriter<>> jsonWriter = TJsonWriterFactory<>::Create(&jsonConfig);
+    FJsonSerializer::Serialize(config.ToSharedRef(), jsonWriter);
+
+    FGenericCrashContext::SetGameData(TEXT("__sentry"), jsonConfig);
+}
+```
+
+You need to call the `ConfigureCrashReporter` some time after your game
+starts up. Any [event attribute](https://develop.sentry.dev/sdk/event-payloads/) can be set.
```
Comment (imatwawana) [inline src/platforms/unreal/setup-crashreport.mdx:118]:
```suggestion
starts. Any [event attribute](https://develop.sentry.dev/sdk/event-payloads/) can be set.
```

File: src/wizard/unreal/index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+---
+name: Unreal Engine
+doc_link: https://docs.sentry.io/platforms/unreal/
+support_level: production
+type: framework
+---
+
+## Gettings Started
```
Comment (imatwawana) [inline src/wizard/unreal/index.md:8]:
```suggestion
## Getting Started
```

**Review by imatwawana — COMMENTED** [review]
Did a preliminary review for wording

**Comment by tustanivsky** [comment]
@imatwawana Thanks for your suggestions, I've updated the PR accordingly

**Comment by bruno-garcia** [comment]
Thank you @tustanivsky , will take a look soon

**Comment by bruno-garcia** [comment]
A few things for us to keep in mind:

- [ ] Make a release so the link shows options for the users to download (https://github.com/getsentry/sentry-unreal/releases)
- [ ] Publish to the Epic Marketplace, point users to that instead of linking to the repo
- [ ] Add the [release-registry](https://github.com/getsentry/sentry-release-registry) so the link to the version and GitHub issues shows on the top right
- [ ] Rename things to `Sentry` instead of `SentrySDK` if possible (is the name free in the Epic store?)
- [ ] We need to be clear that the Unreal Engine Crash Reporter is not preview. It's very stable (supported for 4 years). Only the `uplugin` is preview
- [ ] Docs should start off by clarifying the users must make an option between the plugin or the crash reporter. We should point them to our preference (the plugin) but since it's preview, mentioned the alternative, stable one.
- [ ] The package can take some the copy from the [Sentry SDK in the Unity marketplace](https://assetstore.unity.com/packages/tools/integration/sentry-127333). e.g: `The Sentry's SDK for Unreal Engine allows automatic capture and alert of errors from your Unreal Engine game. Errors in your Blue Prints, C++ script or native crashes make sure you're aware when your users encounter errors, and how to reproduce them to solve problems.`

**Comment by bruno-garcia** [comment]
The current code snippets work for crashes captured by the Crash Reporter: https://sentry-docs-git-fork-tustanivsky-issue-unreal-plugin.sentry.dev/platforms/unreal/enriching-events/context/

Will that end up in crashes captured by the package? If not, can we grab that data somehow and call into the API added to the UE package?

**Comment by tustanivsky** [comment]
> The current code snippets work for crashes captured by the Crash Reporter: https://sentry-docs-git-fork-tustanivsky-issue-unreal-plugin.sentry.dev/platforms/unreal/enriching-events/context/
> 
> Will that end up in crashes captured by the package? If not, can we grab that data somehow and call into the API added to the UE package?

@bruno-garcia This code snippet should work just fine for the events captured by the Sentry plugin. Alternatively, some extra context can be provided to CRC as well (http://localhost:3000/platforms/unreal/setup-crashreport/#configure-attributes)

File: src/platforms/unreal/setup-crashreport.mdx [hunk 5]
```diff
@@ -81,6 +78,59 @@ is enough.
 
 </Alert>
 
+## Configure the Crash Reporter Attributes {#configure-attributes}
+
+Basic attributes can be reconfigured by providing special game data to the
```
Comment (bruno-garcia) [inline src/platforms/unreal/setup-crashreport.mdx:83]: Can we achieve the same through the API in the SDK itself? If so, we could point that out here.

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**